content/post/the-blacker-the-box.md

```diff
@@ -0,0 +1,84 @@
+---
+author: Michael Kaminsky
+title: "The Blacker the Box"
+publishDate: "2018-08-05"
+tags: 
+ - Machine Learning
+ - Data Science
+ - Statistics
+---
+
+There has been a [lot of discussion](https://towardsdatascience.com/machine-learning-how-black-is-this-black-box-f11e4031fdf) in the data science community about the use of black-box models, and there is lots of really [fascinating ongoing](https://arxiv.org/abs/1710.09511) [research](https://storage.googleapis.com/epg-blog-data/epg_2.pdf) into methods, algorithms, [and tools](https://github.com/numeristical/introspective) to help data scientists better introspect their models. While those discussions and that research are important, in this post I will discuss the macro-framework I use for evaluating how black the box can be for a prediction product.
+
+<!--more-->
+
+In this post I will not be getting into the weeds of complexity penalization algorithms or even how to weigh the [tech debt](https://ai.google/research/pubs/pub43146) associated with additional complexity. Instead, I want to take a step back and discuss how I think about "prediction" problems at a more macro level, and how I value accuracy and complexity for different types of problems.
+
+The thesis of this post is:
+
+**The faster the feedback on prediction accuracy, the blacker the box can be. The slower the feedback, the more your models should be explicit and formal.**
+
+In this post I will talk through some examples of fast feedback problems and what makes them more amenable to black-box prediction algorithms (provided you have the proper infrastructure) as well as slower feedback problems and how one might approach predictions in those situations.
+
+## Fast Feedback
+
+The machine learning community spends the bulk of its time working on and talking about fast feedback problems. Problems with fast feedback are defined by 1) the ability to quickly evaluate the correctness of a prediction[^1] and 2) the ability to play the game near infinite amounts of time[^2]. Some examples of fast feedback problems are:
+
+* Chess: it is easy to verify which player has won or lost. Feedback takes only as long as the length of the game.
+* Conversion for an Ad Placement: Feedback to Google or Facebook on whether you clicked a given advertisement, and whether you subsequently converted[^3] is nearly instantaneous.
+* Movie Recommendations: For a given list of potential movies to watch, Netflix gets near instantaneous feedback when you do or do not watch some of the content they have elevated for you.
+
+[Tech-debt considerations](https://ai.google/research/pubs/pub43146) aside, for fast-feedback problems the box can be impenetrably black *as long as you have confidence in your ability to accurately measure the prediction.* 
```

Comment:
You may want to talk about what happens if you can't accurately measure the prediction. I think there are two flavors:
1) You know the right metric to measure, but can't measure it
2) You pick the wrong metric (optimize for click through but forget about conversion)

I think you can go really wrong if you have a type 2 issue

Reply:
I think this is a really important point, but giving this discussion a proper treatment would distract from the post as is. Going to shave this for a future post I think... May also be good for one of @ilanman's  how-to-apply-ML posts.